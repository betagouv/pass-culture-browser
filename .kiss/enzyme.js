import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
