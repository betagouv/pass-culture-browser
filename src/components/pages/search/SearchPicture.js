import PropTypes from 'prop-types'
import React from 'react'

import { ROOT_PATH } from '../../../utils/config'

export const SearchPicture = ({ category }) => (
  <div className="search-picture is-relative">
    <img
      alt={`Rechercher des offres de type ${category}`}
      src={`${ROOT_PATH}/icons/img-${category}.png`}
    />
    <span className="is-absolute text-left fs15">
      {category}
    </span>
  </div>
)

SearchPicture.propTypes = {
  category: PropTypes.string.isRequired,
}
