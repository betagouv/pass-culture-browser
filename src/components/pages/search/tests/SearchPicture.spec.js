import { shallow } from 'enzyme'
import React from 'react'

import SearchPicture from '../SearchPicture'

describe('src | components | pages | search | SearchPicture', () => {
  let props

  beforeEach(() => {
    props = {
      searchType: 'Écouter',
    }
  })

  it('should match snapshot', () => {
    // when
    const wrapper = shallow(<SearchPicture {...props} />)

    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  describe('render()', () => {
    it('should display image with the right url', () => {
      // when
      const wrapper = shallow(<SearchPicture {...props} />)
      const img = wrapper.find('img').props()

      // then
      expect(img.src).toBe('http://localhost/icons/img-Écouter.png')
      expect(img.alt).toBe('Rechercher des offres de type Écouter')
    })
  })
})
