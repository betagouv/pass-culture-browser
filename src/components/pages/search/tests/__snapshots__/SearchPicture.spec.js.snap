// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | search | SearchPicture should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchPicture
    searchType="Écouter"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Rechercher des offres de type Écouter"
          src="http://localhost/icons/img-Écouter.png"
        />,
        <span
          className="is-absolute text-left fs15"
          style={
            Object {
              "backgroundOpacity": 0.75,
            }
          }
        >
          Écouter
        </span>,
      ],
      "className": "search-picture is-relative",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Rechercher des offres de type Écouter",
          "src": "http://localhost/icons/img-Écouter.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Écouter",
          "className": "is-absolute text-left fs15",
          "style": Object {
            "backgroundOpacity": 0.75,
          },
        },
        "ref": null,
        "rendered": "Écouter",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Rechercher des offres de type Écouter"
            src="http://localhost/icons/img-Écouter.png"
          />,
          <span
            className="is-absolute text-left fs15"
            style={
              Object {
                "backgroundOpacity": 0.75,
              }
            }
          >
            Écouter
          </span>,
        ],
        "className": "search-picture is-relative",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Rechercher des offres de type Écouter",
            "src": "http://localhost/icons/img-Écouter.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Écouter",
            "className": "is-absolute text-left fs15",
            "style": Object {
              "backgroundOpacity": 0.75,
            },
          },
          "ref": null,
          "rendered": "Écouter",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
